pkg/cmd/push/cmd.go

```diff
@@ -138,8 +141,17 @@ func runCommand(opts *pushOptions) func(*cobra.Command, []string) error {
 			output.Infof("Pushing %s", opts.srcModelRef.String())
 		}
 		desc, err := PushModel(cmd.Context(), localRepo, remoteRepo, opts)
-		if err != nil {
-			return output.Fatalf("Failed to push: %s. Check you have write permissions to the organization and repository you are pushing to.", err)
+		respErr := &errcode.ErrorResponse{}
+		if ok := errors.As(err, &respErr); ok {
```

Comment:
respErr is already a pointer so is this passing `**errcode.ErrorResponse`?

Reply:
Yeah, this is intended -- since errors.As() sets the value of respErr, it needs a reference to the error itself, which leads to two levels of indirection:

* `errcode.ErrorResponse` is the base struct that stores the data
* `*errcode.ErrorResponse` is a reference to the struct, needed to satisfy the `error` interface (which uses pointer receivers)
* `**errcode.ErrorResponse` is needed to allow errors.As() to set the value of the error; this is required since errors.As() wants a pointer to a type that implements the error interface (in this case, `*errcode.ErrorResponse`).